addons/stock_account/models/stock_quant.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    @api.model
+    def _get_inventory_value(self, quant):
+        if quant.product_id.cost_method in ('real',):
+            return quant.cost * quant.qty
+        return super(StockQuant, self)._get_inventory_value(quant)
+
+    @api.multi
+    def _price_update(self, newprice):
+        ''' This function is called at the end of negative quant reconciliation and does the accounting entries adjustemnts and the update of the product cost price if needed
+        '''
+        super(StockQuant, self)._price_update(newprice)
+        AccountMove = self.env['account.move']
+        for quant in self:
+            move = self._get_latest_move(quant)
+            valuation_update = newprice - quant.cost
+            # this is where we post accounting entries for adjustment, if needed
+            # If neg quant period already closed (likely with manual valuation), skip update
+            if not quant.company_id.currency_id.is_zero(valuation_update) and AccountMove.browse(move.id)._check_lock_date():
```

Comment:
@mdi-odoo,
because "move" is a <b>stock.move</b> and here we need <b>account.move</b>  for calling _check_lock_date method.